nbdime/webapp/nb_server_extension.py

```diff
@@ -8,9 +8,15 @@
 
 from jinja2 import ChoiceLoader, FileSystemLoader
 
-from notebook.utils import url_path_join, to_os_path
-from notebook.services.contents.checkpoints import GenericCheckpointsMixin
-from notebook.services.contents.filecheckpoints import FileCheckpoints
+from jupyter_server.utils import url_path_join, to_os_path
+
+from jupyter_server.services.contents.checkpoints import GenericCheckpointsMixin as jpserver_GenericCheckpointsMixin
+from jupyter_server.services.contents.filecheckpoints import FileCheckpoints as jpserver_FileCheckpoints
+
+from notebook.services.contents.checkpoints import GenericCheckpointsMixin as nbserver_GenericCheckpointsMixin
+from notebook.services.contents.filecheckpoints import FileCheckpoints as nbserver_FileCheckpoints
```

Comment:
I'll try to make these imports guarded so we can have optional dependencies here (and drop the `notebook` dependency from setup.py).